specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-08-01-preview/IncidentCreationRules.json

```diff
@@ -0,0 +1,640 @@
+          "description": "Determines the lookback time frame for which grouping will be applied.",
+          "type": "string"
+        },
+        "groupingKind": {
+          "$ref": "#/definitions/GroupingKind"
+        },
+        "reopenClosedIncident": {
+          "description": "Determines if a closed incident should be reopened in case the grouping properties are matched.",
+          "type": "boolean"
+        }
+      },
+      "discriminator": "groupingKind"
+    },
+    "IncidentCreationRuleProperties": {
+      "description": "Incident creation rule properties.",
+      "required": [
+        "createIncident",
+        "displayName",
+        "enabled",
+        "order"
+      ],
+      "type": "object",
+      "properties": {
+        "displayName": {
+          "description": "The display name of the incident creation rule.",
+          "minLength": 1,
+          "type": "string",
+          "maxLength": 500
+        },
+        "description": {
+          "description": "The description of the incident creation rule.",
+          "type": "string",
+          "maxLength": 1500
+        },
+        "order": {
+          "format": "int32",
+          "description": "The order of evaluation of the incident creation rule",
+          "type": "integer",
+          "minimum": 1,
+          "maximum": 1000
+        },
+        "conditions": {
+          "description": "The conditions to evaluate to determine if the incident creation rule should be applied.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IncidentCreationRuleCondition"
+          },
+          "x-ms-identifiers": []
+        },
+        "createIncident": {
+          "description": "Determines whether to create an incident based on this rule or not.",
+          "type": "boolean"
+        },
+        "groupingConfiguration": {
+          "$ref": "#/definitions/IncidentCreationRuleGroupingConfiguration"
+        },
+        "enabled": {
+          "description": "Determines whether the incident creation rule is enabled or disabled.",
+          "type": "boolean"
+        },
+        "lastModifiedTimeUtc": {
+          "format": "date-time",
+          "description": "The last time the incident creation rule was updated.",
+          "type": "string",
+          "readOnly": true
+        },
+        "createdTimeUtc": {
+          "format": "date-time",
+          "description": "The time the incident creation rule was created.",
+          "type": "string",
+          "readOnly": true
+        },
+        "lastModifiedBy": {
+          "type": "object",
+          "$ref": "../../../common/2.0/types.json#/definitions/ClientInfo"
+        },
+        "createdBy": {
+          "type": "object",
+          "$ref": "../../../common/2.0/types.json#/definitions/ClientInfo"
+        }
+      }
+    },
+    "IncidentCreationRulePropertyBasedGroupingProperties": {
+      "type": "object",
+      "properties": {
+        "groupingAttributes": {
```

Comment:
>groupingAttributes [](http://example.com/codeflow?start=9&length=18)

You have groupingProperties.groupingAttributes[ ... ]. This adds unnecessary and confusing nesting. I'd recommend eliminating one level of nesting and use consistent terminology ("properties" seems better than "attributes", since the array contains "propertyName"-"propertyValue" pairs). Maybe even simplify it to something more like groupingValues[ { name: a, value x }, { name: b, value y }, ... ]?